tiny_cnn/core/kernels/avx_conv2d_back_kernel.h

```diff
@@ -0,0 +1,626 @@
+
+            idx = params.out.get_index(0, 0, outc);
+            const float_t *pdelta_src = &curr_delta[idx];
+
+            idx = params.in_padded.get_index(0, 0, inc);
+            float_t *pdelta_dst = &(*prev_delta)[idx];
+
+            for (cnn_size_t y = 0; y < params.out.height_; y++) {
+                for (cnn_size_t x = 0; x < params.out.width_; x++) {
+                    const float_t * ppw = pw;
+
+                    idx = y * params.out.width_ + x;
+                    const float_t ppdelta_src = pdelta_src[idx];
+
+                    float_t * ppdelta_dst = pdelta_dst +
+                          y * params.h_stride * params.in_padded.width_ +
+                          x * params.w_stride;
+
+                    for (cnn_size_t wy = 0; wy < 5 /* params.weight.height_ */; wy++) {    // NOLINT
+                        for (cnn_size_t wx = 0; wx < 5 /* params.weight.width_ */; wx++) { // NOLINT
+                            idx = wy * params.in_padded.width_ + wx;
+                            ppdelta_dst[idx] += *ppw++ * ppdelta_src;
+                        }
+                    }
+                }
+            }
+        }
+    });
+
+    // accumulate dw
+    for_i(params.in.depth_, [&](int inc) {
+        for (cnn_size_t outc = 0; outc < params.out.depth_; outc++) {
+            if (!params.tbl.is_connected(outc, inc)) continue;
+
+            for (cnn_size_t wy = 0; wy < 5 /* params.weight.height_ */; wy++) {
+                for (cnn_size_t wx = 0; wx < 5 /* params.weight.width_ */; wx++) {
+                    float_t dst = float_t(0);
+
+                    cnn_size_t idx = 0;
+                    idx = params.in_padded.get_index(wx, wy, inc);
+                    const float_t * prevo = &prev_out[idx];
+
+                    idx = params.out.get_index(0, 0, outc);
+                    const float_t * delta = &curr_delta[idx];
+
+                    for (cnn_size_t y = 0; y < params.out.height_; y++) {
+                        dst += vectorize::dot(
+                            prevo + y * params.in_padded.width_,
+                            delta + y * params.out.width_,
+                            params.out.width_);
+                    }
+
+                    idx = params.in.depth_ * outc + inc;
+                    dW[params.weight.get_index(wx, wy, idx)] += dst;
+                }
+            }
+        }
+    });
+
+    // accumulate db
+    if (params.has_bias) {
+        //vec_t& db = *in_grad[2];
+
+        for (cnn_size_t outc = 0; outc < params.out.depth_; outc++) {
+            cnn_size_t idx = params.out.get_index(0, 0, outc);
+            const float_t * delta = &curr_delta[idx];
+            const float_t * deltaa = delta + params.out.width_ *
+                                             params.out.height_;
+            db[outc] += std::accumulate(delta, deltaa, float_t(0));
+        }
+    }
+} // avx_conv2d_back_5x5_kernel double ver
+
+void avx_conv2d_back_kernel(const conv_params& params,
```

Comment:
Should I put `inline` keyword into above the line of its prototype or the same line?